docs/import/import_data.js

```diff
@@ -140,132 +135,327 @@ function stringifyGeoid(feature) {
+  return true;
 }
 
+const damageError = {
+  damage: null,
+  damageEnvelope: null,
+};
+
+// Distance in meters away from damage point that we are still interested in
+// collecting information.
+const damageBuffer = 1000;
+
 /**
- * Performs operation of processing inputs and creating output asset.
- * @param {Promise} firebaseAuthPromise
+ * Calculates damage if there is a damage asset, or simply writes the
+ * user-provided bounds to Firestore if not.
+ * @param {Object} assetData
+ * @return {{damage: ?ee.FeatureCollection, damageEnvelope:
+ *     ee.Geometry.Rectangle}|{damage: null, damageEnvelope: null}} Returns
+ *     the damage asset (if present) and the envelope bounding the damage, or
+ *     both null if an error occurs
  */
-function run(firebaseAuthPromise) {
-  const resources = getResources();
-  const damage = ee.FeatureCollection(resources.damage);
-  const centerStatusLabel = document.createElement('span');
-  centerStatusLabel.innerText = 'Computing and storing bounds of map: ';
+function calculateDamage(assetData) {
+  const damagePath = assetData['damage_asset_path'];
   const centerStatusSpan = document.createElement('span');
+  const centerStatusLabel = document.createElement('span');
+  $('.compute-status').append(centerStatusLabel).append(centerStatusSpan);
   centerStatusSpan.innerText = 'in progress';
-  storeCenter(damage, firebaseAuthPromise)
-      .then(trimGeoNumbers)
-      .then(
-          (bounds) => centerStatusSpan.innerText = 'Found bounds (' +
-              bounds[0] + ', ' + bounds[1] + '), (' + bounds[2] + ', ' +
-              bounds[3] + ')')
-      .catch((err) => centerStatusSpan.innerText = err);
-  $('.compute-status').append(centerStatusLabel);
-  $('.compute-status').append(centerStatusSpan);
+  const firestoreError = (err) => centerStatusSpan.innerText +=
+      'Error writing bounds to Firestore: ' + err;
+  if (damagePath) {
+    const damage = ee.FeatureCollection(damagePath);
+    // Uncomment to test with a restricted damage set (14 block groups' worth).
+    // damage = damage.filterBounds(
+    //     ee.FeatureCollection('users/gd/2017-harvey/data-ms-as-nod')
+    //         .filterMetadata('GEOID', 'starts_with', '482015417002'));
+    centerStatusLabel.innerText = 'Computing and storing bounds of map: ';
+    storeCenter(damage)
+        .then(displayGeoNumbers)
+        .then((bounds) => centerStatusSpan.innerText = 'Found bounds ' + bounds)
+        .catch((err) => centerStatusSpan.innerText = err);
+    return {damage, damageEnvelope: damage.geometry().buffer(damageBuffer)};
+  }
+  // TODO(janakr): in the no-damage case, we're storing a rectangle, but
+  //  experiments show that, at least for Harvey, the page is very slow when we
+  //  load the entire rectangle around the damage. Maybe allow users to select a
+  //  polygon so they can draw a tighter area?
+  centerStatusLabel.innerText = 'Storing bounds of map: ';
+  const damageSw = assetData['map_bounds_sw'];
```

Comment:
maybe this should also go under map-bounds not asset-data?

Reply:
My conception is that asset_data is about the raw ingredients that go into the final data that the map uses, while map-bounds is that final data. So I think some separation is good.